ldap/servers/plugins/uiduniq/uid.c

```diff
@@ -1392,13 +1392,13 @@ preop_modrdn(Slapi_PBlock *pb)
              */
             if (NULL != markerObjectClass) {
                 /* Subtree defined by location of marker object class */
-                result = findSubtreeAndSearch(slapi_entry_get_sdn(e), attrNames, attr, NULL,
-                                              requiredObjectClass, superior,
+                result = findSubtreeAndSearch(superiorSDN, attrNames, attr, NULL,
+                                              requiredObjectClass, sourceSDN,
                                               markerObjectClass, config->exclude_subtrees);
             } else {
                 /* Subtrees listed on invocation line */
                 result = searchAllSubtrees(config->subtrees, config->exclude_subtrees, attrNames, attr, NULL,
-                                           requiredObjectClass, superior, config->unique_in_all_subtrees);
+                                           requiredObjectClass, superiorSDN, sourceSDN, config->unique_in_all_subtrees);
```

Comment:
Interface is searchAllSubtrees(.., source, destination,..), shouldn't it be searchAllSubtrees(..., sourceSDN, "<rdn>,<superiorSDN>",... )

Reply:
Same as above. We need sourceSDN so we can ignore it in the results.

Reply:
Your change looks valid to me, using destinationSDN and sourceSDN